### ISSUE TITLE
How to load own dataset to train rather than only mnist dataset？

### ISSUE THREAD

**Issue by Ronales** [adambielski/siamese-triplet#33]
Dear author ，I want to use my dataset to train the model，but I think the load way of mnist can't suit me，Can you tell me how to change the load my dataset？ thanks！

**Comment by adambielski** [comment]
Read up on how to use dataset obejcts and dataloader in PyTorch.
Also check issue #12

**Comment by Lovebubu** [comment]
I use my own data which contains similar pairs and dissimilar pairs. For example, I would like to use Siamese network. 
<pre>
## load my own data, the item of the my dataset is [(img0,img1),1]
totaldatasets = loadPickle("pairsImgLabel_112.pickle")
shuffle(totaldatasets)

## get train data and test data
train_len = 8000
train_data, test_data = torch.utils.data.random_split(totaldatasets, [train_len, len(totaldatasets) - train_len])

## write my own dataset class
Class RuiDataset(Dataset):
    def __init__(self, heatmap_dataset, train, transform=None):
        self.heatmap_dataset = heatmap_dataset
        self.transform = transform
        self.train = train
        
        if self.train:
            self.train_data = self.heatmap_dataset
        else:
            self.test_data = self.heatmap_dataset

    def __len__(self):
        return len(self.heatmap_dataset)

    def __getitem__(self, idx):
        img1 = self.heatmap_dataset[idx][0][0]
        img2 = self.heatmap_dataset[idx][0][1]
        target = self.heatmap_dataset[idx][1]
        if self.transform is not None:
            img1 = self.transform(img1)
            img2 = self.transform(img2)
        return (img1, img2), target

## generate the data similar to the SiameseMNIST(Dataset), then put the dataset to dataloader to train model
mean, std = 0.28604059698879553, 0.35302424451492237
transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize((mean,), (std,))
])

rui_train_data = RuiDataset(train_data, True, transform)
rui_test_data = RuiDataset(test_data, False, transform)

batch_size = 20
kwargs = {'num_workers': 1, 'pin_memory': True} if cuda else {}
siamese_train_loader = torch.utils.data.DataLoader(
    rui_train_data, batch_size=batch_size, shuffle=True, **kwargs)
siamese_test_loader = torch.utils.data.DataLoader(
    rui_test_data, batch_size=batch_size, shuffle=False, **kwargs)
print(len(siamese_train_loader))
</pre>
